GitCommands/Git/GitRevision.cs

```diff
@@ -94,6 +94,12 @@ public BuildInfo BuildStatus
 
         public string Name { get; set; }
 
+        /// <summary>
+        /// Used to capture the other output from log calls.  Ex: getting commit history from a file using --follow and --name-only gives us the filename.
+        /// </summary>
+        [CanBeNull]
+        public string AdditionalOutput { get; set; }
```

Comment:
Apparently, the `Name` property is meant to hold the filename, so is `AdditionalOutput` really required? You can check out https://github.com/gitextensions/gitextensions/blob/release/2.51/GitCommands/RevisionGraph.cs#L403 and https://github.com/gitextensions/gitextensions/blob/release/2.51/GitCommands/RevisionGraph.cs#L321 to see how it was used before one of the refactorings that killed the `--follow` feature.

Reply:
(I guess if you leave the `AdditionalOutput`, you should probably remove the `Name` property, as it's no longer assigned.)

Reply:
I named it AdditionalOutput because that is exactly what that is.  The AdditionalOutput is then processed based on what output it is.  In this case name will eventually be filled in but Other commands could do this to.  So I will be leaving name and additional output so that it leaves a spot for all commands to process the additional output after the null.

Reply:
Good point. Then probably just remove the `GitRevision.Name` property as it's never assigned any more?

Reply:
I don't think you get the idea. AditionalOutput is the basically raw output.  This is low level code. You can't assume it's always filenames. I named it to be quite clear on what it was. Other commands may create additional output that is not filenames or other info. So eventually in this case name will be set up but I can't 100% say additional output is always filenames. So I will not be removing name or AditionalOutput.

Reply:
Look. This is wip I haven't finished. I need parsing done first.

Reply:
Maybe outline what the plan and the vision are, it could help in understanding and remove some questions

Reply:
> I don't think you get the idea

I think I do get the idea and, as I said, I think you have a good point.

However. `GitRevision.Name` used to contain the filename back in 2.50, when rename history still worked. It is now currently always `null`, due to [one of the two refactorings that broke rename file history](https://github.com/gitextensions/gitextensions/commit/1f6b0cacc8bc78be076b4e706e5aa030b6d36ea0). Your fix leaves it at `null` and introduces `AdditionalInfo` instead.

The property is still read in over 100 places, though, which might still be broken. That's why I suggested restoring the old behavior and using `Name` even though it's not always the filename. And if you don't want to do this - as I said, you're making a good argument, but one should still think about what to do about the `Name` property.

But getting the parsing done has higher priority, I see, np.